File: libnetwork/src/main/java/com/gystry/libnetwork/JsonCallback.java
package com.gystry.libnetwork;

/**
 * @author gystry
 * 创建日期：2020/7/3 17
 * 邮箱：gystry@163.com
 * 描述：
 */
public abstract class JsonCallback<T> {
    public void onSuccess(ApiResponse<T> response) {
    }

    public void onError(ApiResponse<T> response) {
    }

    public void onCacheSuccess(ApiResponse<T> response) {
    }
}

File: libnetworkkt/src/main/java/com/gystry/libnetworkkt/JsonCallback.kt
package com.gystry.libnetworkkt

/**
 * @author gystry
 * 创建日期：2021/3/18 11
 * 邮箱：gystry@163.com
 * 描述：
 */
abstract class JsonCallback<T> {
    open fun onSuccess(response:ApiResponse<T>?){}
    open fun onError(response:ApiResponse<T>?){}
    open fun onCacheSuccess(response:ApiResponse<T>){}
}